src/Adapter/Customer/Group/Validate/CustomerGroupValidator.php

```diff
@@ -0,0 +1,56 @@
+<?php
+/*
+ * Copyright since 2007 PrestaShop SA and Contributors
+ * PrestaShop is an International Registered Trademark & Property of PrestaShop SA
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.md.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/OSL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to https://devdocs.prestashop.com/ for more information.
+ *
+ * @author    PrestaShop SA and Contributors <contact@prestashop.com>
+ * @copyright Since 2007 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ */
+
+namespace PrestaShop\PrestaShop\Adapter\Customer\Group\Validate;
+
+use Group as CustomerGroup;
+use PrestaShop\PrestaShop\Adapter\AbstractObjectModelValidator;
+use PrestaShop\PrestaShop\Adapter\Shop\Repository\ShopRepository;
+
+class CustomerGroupValidator extends AbstractObjectModelValidator
+{
+    /**
+     * @var ShopRepository
+     */
+    private $shopRepository;
+
+    public function __construct(ShopRepository $shopRepository)
+    {
+        $this->shopRepository = $shopRepository;
+    }
+
+    public function validateCreation(CustomerGroup $customerGroup)
```

Comment:
```suggestion
    public function validateCreation(CustomerGroup $customerGroup): void
```

Reply:
Does creation differ from edition?
I guess we will see in other PR where you handle edition anyway.

Reply:
```suggestion
    public function validate(CustomerGroup $customerGroup)
```
I'm not sure the validation will be different for creation and edition, maybe you can call it a generic `validate` and when you tackle with the update command and realize the process really need to be split then you can split into two validation methods

Reply:
Done